device/transport/amqp/src/amqp.ts

```diff
@@ -283,7 +287,12 @@ export class Amqp extends EventEmitter implements DeviceTransport {
                   /*Codes_SRS_NODE_DEVICE_AMQP_13_002: [ The connect method shall set the CA cert on the options object when calling the underlying connection object's connect method if it was supplied. ]*/
                   if (this._options && this._options.ca) {
                     config.sslOptions = config.sslOptions || {};
-                    config.sslOptions.caFile = this._options.ca;
+                    /*Codes_SRS_NODE_DEVICE_AMQP_06_012: [The `connect` method shall first test if the `ca` property is the name of an already existent file.  If so, it will attempt to read that file as a pem into a string value and pass the string to config object `ca` property.  Otherwise, it is assumed to be a pem string.] */
+                    if (fs.existsSync(this._options.ca)) {
+                      config.sslOptions.ca = fs.readFileSync(this._options.ca, 'utf8');
```

Comment:
>config.sslOptions.ca = fs.readFileSync(this._options.ca, 'utf8'); [](start = 22, length = 65)

And, for what it's worth, I'm not sure where this requirement came from?  I don't think MQTT or HTTP supports the readFile functionality?  I think they only accept strings.  (Of course, I've been wrong about stuff like this plenty of times before....)

Reply:
history of the amqp10 library

---
In reply to: [214414741](https://github.com/Azure/azure-iot-sdk-node/pull/354#discussion_r214414741) [](ancestors = 214414741)

Reply:
Yup, it's the way things were with the previous incantation.  This really seems like the least bad solution.

---
In reply to: [214525410](https://github.com/Azure/azure-iot-sdk-node/pull/354#discussion_r214525410) [](ancestors = 214525410,214414741)